### PR TITLE
Insticator Bid Adapter: Add support for BidFloors

### PR DESCRIPTION
**Pull request by shubhamc-ins**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

File: modules/insticatorBidAdapter.js [hunk 1]
```diff
@@ -178,6 +190,48 @@ function buildImpression(bidRequest) {
...
+      moduleBidFloor = bidRequest.getFloor({
+        mediaType: _mediaType,
+        size: _size
+      });
+    } catch (err) {
+      // continue with no module floors
+      logWarn('priceFloors module call getFloor failed, error : ', err);
+    }
+
+    if (moduleBidFloor && moduleBidFloor.currency === 'USD') {
```
Comment (muuki88) [inline modules/insticatorBidAdapter.js:229]: prebid offers a currency conversion module. With this code you effectively disable floors for everybody that doesn't use USD, even though it should work.

See https://docs.prebid.org/dev-docs/modules/floors.html#currency

**Comment by muuki88** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/5307

File: modules/insticatorBidAdapter.js (hunk 1 above)
Comment (shubhamc-ins) [reply modules/insticatorBidAdapter.js]: @muuki88 Updated the PR. Please check now.

I have added the default currency as `USD`.

Comment (muuki88) [reply modules/insticatorBidAdapter.js]: Nice 👍 LGTM

**Comment by shubhamc-ins** [comment]
> Docs PR: [prebid/prebid.github.io#5307](https://github.com/prebid/prebid.github.io/pull/5307)

@muuki88 As I updated the logic in this PR. I think we don't need any changes in the Docs PR. Let me know what do you think.

File: modules/insticatorBidAdapter.js [hunk 2]
```diff
@@ -178,6 +190,49 @@ function buildImpression(bidRequest) {
...
+    }
+    try {
+      moduleBidFloor = bidRequest.getFloor({
+        currency: 'USD',
+        mediaType: _mediaType,
+        size: _size
+      });
+    } catch (err) {
+      // continue with no module floors
+      logWarn('priceFloors module call getFloor failed, error : ', err);
```
Comment (ncolletti) [inline modules/insticatorBidAdapter.js:227]: Could you add unit tests to cover this function's expected behavior with various mediatypes and size parsing?

File: modules/insticatorBidAdapter.js [hunk 3]
```diff
@@ -170,6 +170,18 @@ function buildImpression(bidRequest) {
...
+  let bidFloor = parseFloat(deepAccess(bidRequest, 'params.floor'));
+
+  if (!isNaN(bidFloor)) {
+    imp.bidfloor = deepAccess(bidRequest, 'params.floor');
+    imp.bidfloorcur = 'USD';
+    const bidfloorcur = deepAccess(bidRequest, 'params.bidfloorcur')
+    if (bidfloorcur && bidfloorcur !== 'USD') {
+      delete imp.bidfloor;
+      delete imp.bidfloorcur;
+    }
```
Comment (ncolletti) [inline modules/insticatorBidAdapter.js:182]: Please include the new params `bidfloor` and `bidfloorcur` to your markdown file. A mention that only `USD` is accepted would be useful, along with a logWarn here

File: modules/insticatorBidAdapter.js (hunk 2 above)
Comment (shubhamc-ins) [reply modules/insticatorBidAdapter.js]: @ncolletti Added the test cases.

Please let me know if this looks good now.

File: modules/insticatorBidAdapter.js (hunk 3 above)
Comment (shubhamc-ins) [reply modules/insticatorBidAdapter.js]: Updated the description in Docs https://github.com/prebid/prebid.github.io/pull/5307 PR.

**Review by ncolletti — APPROVED** [review]
thanks for the updates, looks good!

**Pull request merged**